Mobile Buy SDK/Mobile Buy SDK/Operations/BUYRequestOperation.m

```diff
@@ -169,14 +168,24 @@ - (NSURLSessionDataTask *)request:(NSURLRequest *)request completion:(BUYRequest
 			json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
 		}
 		
-		NSHTTPURLResponse *httpResponse = (id)response;
-		if (!httpResponse.successful && !error) {
-			error = [[NSError alloc] initWithDomain:kShopifyError code:httpResponse.statusCode userInfo:json];
+		NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+		if (!error) {
+			error = [self errorFromJSON:json response:httpResponse];
 		}
 		
 		completion(json, httpResponse, error);
 	}];
 	return task;
 }
 
+#pragma mark - Error -
+
+- (NSError *)errorFromJSON:(NSDictionary *)json response:(NSHTTPURLResponse *)response
```

Comment:
This would seem less weird if the response code was provided as the first argument.

Although, it also seems odd to wrap the body of a method, even a one-liner, with a conditional. Usually it's the job of the caller to perform conditional checks. Ideally, methods always do what they say they do, not conditionally. I think I liked the original design a bit better.